### PR TITLE
Add ordering and forecasting.

### PR DESCRIPTION
**Pull request by stbof**
DOC-5335
DOC-5334

### CODE REVIEW

**Review by michae2 — APPROVED** [review]
:lgtm: Looks good! Thank you!

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/15177#-:-NCawhZjC3Z8yQoX9MrF:bytkqq7)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @stbof)

___
*[`v22.2/show-statistics.md` line 83 at r1](https://reviewable.io/reviews/cockroachdb/docs/15177#-NCawRe77sRFhaX_CPLD:-NCawRe853BbWcFgFNNF:bb4po1e) ([raw file](https://github.com/cockroachdb/docs/blob/34f41343d8d9d075c44c00107169c59e78514bae/v22.2/show-statistics.md#L83)):*
> ```Markdown
> {% include_cached copy-clipboard.html %}
> ~~~ sql
> > SHOW STATISTICS FOR TABLE rides with FORECAST;
> ```

nit: I would make "with" all-caps here, since it's a SQL keyword.

File: v22.2/show-statistics.md [hunk 1]
```diff
@@ -51,22 +51,96 @@ Parameter      | Description
...
+CockroachDB generates forecasted statistics when a couple of conditions are met:
+
+- There need to be at least 3 historical statistics collections.
+- The historical statistics need to closely fit a linear pattern.
+
+The following example shows 3 historical statistics collections and the subsequent forecast.
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+> SHOW STATISTICS FOR TABLE rides with FORECAST;
```
Comment (stbof) [inline v22.2/show-statistics.md:83]:
```suggestion
> SHOW STATISTICS FOR TABLE rides WITH FORECAST;
```

**Comment by stbof** [comment]
> :lgtm: Looks good! Thank you!
> 
> > Reviewed 1 of 1 files at r1, all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/15177#-:-NCawhZjC3Z8yQoX9MrF:bytkqq7)_ status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @stbof)
> 
> _[`v22.2/show-statistics.md` line 83 at r1](https://reviewable.io/reviews/cockroachdb/docs/15177#-NCawRe77sRFhaX_CPLD:-NCawRe853BbWcFgFNNF:bb4po1e) ([raw file](https://github.com/cockroachdb/docs/blob/34f41343d8d9d075c44c00107169c59e78514bae/v22.2/show-statistics.md#L83)):_
> 
> > ```gfm
> > {% include_cached copy-clipboard.html %}
> > ~~~ sql
> > > SHOW STATISTICS FOR TABLE rides with FORECAST;
> > ```
> 
> nit: I would make "with" all-caps here, since it's a SQL keyword.

Fixed. Thanks for the catch.

File: v22.2/show-statistics.md [hunk 2]
```diff
@@ -51,22 +51,96 @@ Parameter      | Description
...
+  __auto__        | {end_time}                | 2022-09-22 16:45:29.957103 |       500 |            367 |          0 |        7 | 798866571269537793
+  __auto__        | {revenue}                 | 2022-09-22 16:45:29.957103 |       500 |            100 |          0 |        6 | 798866571283103745
 (13 rows)
 ~~~
 
+### Show forecast statistics
+
+The `WITH FORECAST` option calculates and displays forecasted statistics along with the existing table statistics. The forecast is a simple regression model that predicts how the statistics have changed since they were last collected.
+
+CockroachDB generates forecasted statistics when a couple of conditions are met:
```
Comment (taroface) [inline v22.2/show-statistics.md:74]:
```suggestion
CockroachDB generates forecasted statistics when the following conditions are met:
```

File: v22.2/show-statistics.md [hunk 3]
```diff
@@ -51,22 +51,96 @@ Parameter      | Description
...
+
+- There need to be at least 3 historical statistics collections.
+- The historical statistics need to closely fit a linear pattern.
+
+The following example shows 3 historical statistics collections and the subsequent forecast.
```
Comment (taroface) [inline v22.2/show-statistics.md:79]:
```suggestion
The following example shows 3 historical statistics collections and the subsequent forecast:
```

**Review by taroface — APPROVED** [review]
LGTM; couple nits.

**Pull request merged**